checker/src/call_visitor.rs

```diff
@@ -2137,7 +2137,17 @@ impl<'call, 'block, 'analysis, 'compilation, 'tcx, E>
                     }
                 }
 
-                // The tag type should be a generic ADT whose first parameter is a constant.
+                // The tag type could be the pre-defined ConstantTimeTaint type in mirai_annotations.
+                if let TyKind::Adt(adt_def, ..) = tag_rustc_type.kind {
+                    if self.block_visitor.bv.tcx.def_path_str(adt_def.did)
```

Comment:
This is expensive. Rather use KnownNames for this. Better yet, don't single out this type at all.